cpp/code/CMakeLists.txt

```diff
@@ -42,9 +49,14 @@ function(RECIPE TARGET)
             common.cc
             main.cc
     )
+    if(TARGET Arrow::arrow_shared)
+        set(ARROW_SHARED_NAME Arrow::arrow_shared)
+    else()
+        set(ARROW_SHARED_NAME arrow_shared)
+    endif()
     target_link_libraries(
             ${TARGET}
-            arrow_shared
+            ${ARROW_SHARED_NAME}
             arrow_dataset
             arrow_flight
             parquet
```

Comment:
Ah, sorry. I missed this.

CMake packages for Apache Arrow 10.0.0 have correct dependency information. So we don't need to all Arrow related CMake targets explicitly:

```cmake
if(TARGET Arrow::arrow_shared)
  target_link_libraries(${TARGET} ArrowDataset::arrow_dataset_shared ArrowFlight::arrow_flight_shared gtest)
else()
  target_link_libraries(parquet_shared arrow_shared)
  target_link_libraries(arrow_dataset_shared parquet_shared)
  target_link_libraries(arrow_flight_shared arrow_shared)
  target_link_libraries(${TARGET} arrow_dataset_shared arrow_flight_shared gtest)
endif()

```

Reply:
I had to use the `INTERFACE` keyword here:
```
        target_link_libraries(parquet_shared INTERFACE arrow_shared)
        target_link_libraries(arrow_dataset_shared INTERFACE parquet_shared)
        target_link_libraries(arrow_flight_shared INTERFACE arrow_shared)
```
otherwise I got a bunch of errors:
```
CMake Error at CMakeLists.txt:60 (target_link_libraries):
  IMPORTED library can only be used with the INTERFACE keyword of
  target_link_libraries
Call Stack (most recent call first):
  CMakeLists.txt:75 (recipe)
```

Reply:
Oh, sorry. Yes. I missed `INTERFACE`...